plugins/core/gov.nasa.jpl.imce.oml.viewpoint/src/gov/nasa/jpl/imce/oml/viewpoint/AspectConceptClassDiagramService.xtend

```diff
@@ -0,0 +1,146 @@
+package gov.nasa.jpl.imce.oml.viewpoint
+
+import gov.nasa.jpl.imce.oml.model.terminologies.AspectSpecializationAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.Concept
+import gov.nasa.jpl.imce.oml.model.terminologies.ConceptSpecializationAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.Entity
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityExistentialRestrictionAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityRelationship
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityRestrictionAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.ReifiedRelationship
+import gov.nasa.jpl.imce.oml.model.terminologies.SpecializationAxiom
+import java.util.HashSet
+import java.util.Set
+import org.eclipse.sirius.diagram.DDiagram
+import org.eclipse.sirius.diagram.DSemanticDiagram
+
+class AspectConceptClassDiagramService {
+	
+	/*
+	 * Gets root {@link Entity} which the passed {@link DDiagram}
+	 * was created from
+	 * 
+	 * @param The diagram
+	 * @return The root {@link Entity}
+	 */
+	def Entity getRootEntity(DDiagram d){
+		return (d as DSemanticDiagram).target as Entity
+	}
+
+     /*
+	 * Gets all {@link EntityRelationship}s with the 
+	 * passed {@link} Concept as its relation domain
+	 * 
+	 * @param c The root Concept
+	 * @return Set of {@link ReifiedRelationship}s
+	 */
+	def Set<EntityRelationship> getVisualRelationshipsWithRootAsDomain(Entity e){
+		return e.tbox.boxStatements.
+		filter(EntityRelationship).
+		filter[f | f.source == e].
+		toSet
+	}
+	
+	/*
+	 * Gets all {@link ReifiedRelationship}s with the 

```

Comment:
This particular diagram will not.  It Should be clear now with the added class comments.